config/class-sync.php

```diff
@@ -12,6 +12,11 @@ class Sync {
 
 	const JETPACK_PRIVACY_SETTINGS_SYNC_STATUS_OPTION_NAME = 'vip_config_jetpack_privacy_settings_synced_value';
 
+	/**
+	 * @var array List of blogIDs that need a sync
+	 */
+	private $blogs_to_sync = [];
```

Comment:
If we use an associative array `[ $blog_id => $blog_id,  $blog_2_id => $blog_2_id ]`  we can reduce check's complexity to O(1): 
instead of `array_search` or `is_array` this becomes a simple `isset( $this->blog_ids_to_sync[ get_current_blog_id() ] )`

Reply:
This list should never be big, in fact I'm not sure that it will ever be more than 1 - at least through core functionality. It's either the blog you are on, or another blog being updated in network admin. Pretty sure the only way it grows is through custom code that loops through blogs and does an update.

Even in that case though, with 10k entries we're still talking milliseconds. I'd argue if it's worth the complexity & memory difference. But also, neither is large enough to matter either 😆

Reply:
@rinatkhaziev I don't have strong opinions on either implementation. Is this a blocker for you?

Reply:
yikes, missed I this comment, sorry! @andrea-sdl